### PR TITLE
Adds the Jetpack banner to Stats

### PR DESCRIPTION
**Pull request by guarani**
This PR adds the Jetpack banner to the Stats screens. The banner is a sticky footer, so it stays fixed to the bottom of the screen as the user scrolls content.

This PR is based on the work in https://github.com/wordpress-mobile/WordPress-iOS/pull/18977.

To test:

1. Banner should not appear in Jetpack app
2. Banner should appear on all top-level Stats screens
3. Banner should look good in Dark Mode
4. Banner should look good on iPad

### Screenshots

| Light mode | Dark mode |
| - | - |
| Screen Shot 2022-07-01 at 19 56 26 | Screen Shot 2022-07-01 at 19 56 09 |

| iPad portrait | iPad landscape |
| - | - |
| Screen Shot 2022-07-01 at 20 00 16 | Screen Shot 2022-07-01 at 20 00 00 |

## Regression Notes

1. Potential unintended areas of impact

The Stats screen is the only area of the app impacted by this change. The most likely issue would be that the banner somehow interferes with content on the screen.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing of the stats screen

3. What automated tests I added (or what prevented me from doing so)

I just relied on automated tests to ensure the core flows of the app are not affected.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    func setup() {
+        guard AppConfiguration.isWordPress else {
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift]: Not sure if the responsibility of where the view is used/configured should be of the view itself, what do you think?

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboard.storyboard [hunk 2]
```diff
@@ -41,19 +41,29 @@
...
                             </view>
                             <containerView opaque="NO" contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="J6j-x0-sMS">
-                                <rect key="frame" x="0.0" y="90" width="375" height="577"/>
+                                <rect key="frame" x="0.0" y="90" width="375" height="557"/>
                                 <connections>
                                     <segue destination="83f-eK-Wfs" kind="embed" id="Vur-EF-t51"/>
                                 </connections>
                             </containerView>
+                            <view contentMode="scaleToFill" verticalHuggingPriority="750" translatesAutoresizingMaskIntoConstraints="NO" id="zMO-yu-sCx" customClass="JetpackBannerView" customModule="WordPress" customModuleProvider="target">
+                                <rect key="frame" x="0.0" y="647" width="375" height="20"/>
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Stats/SiteStatsDashboard.storyboard:50]: From the designs, the height of the banner should be 44. At runtime, it's also different (see below)
<p align=center>

</p>

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            return
+        }
+
+        translatesAutoresizingMaskIntoConstraints = false
+        backgroundColor = Appearance.jetpackBackgroundColor
+
+        let jetpackButton = makeJetpackButton()
+        addSubview(jetpackButton)
+
+        pinSubviewToAllEdges(jetpackButton, insets: UIEdgeInsets(top: 5, left: 0, bottom: 5, right: 0), priority: .required)
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift]: Why do we need to add insets to the button? And if we do, could they be added to the `Appearance` enum?

**Review by Gio2018 — COMMENTED** [review]
Hey @Guarani , I tested it and it pretty much worked. I found a sizing misalignment with the designs, plus a couple more comments.
Thank you!

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift (hunk 1 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift]: That's a good point, I'll see if it's feasible to move the logic here to the Stats controller. Doing so might create duplication if this is used in other areas, but I'm also wary of having the logic in the view itself.

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboard.storyboard (hunk 2 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Stats/SiteStatsDashboard.storyboard:50]: Thanks for spotting that discrepancy, will fix!

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift (hunk 3 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift]: I added those insets to get close to the desired 44px height. I might try just making the button the full height though. 
Thanks for flagging.

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboard.storyboard (hunk 2 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Stats/SiteStatsDashboard.storyboard:50]: Fixed now, it was 6px too short. Updated now to 44px.

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift (hunk 1 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift]: I moved the logic to the Stats dashboard view controller 👍

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift (hunk 3 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift]: I've removed these, thanks!

**Comment by guarani** [comment]
I'm re-requesting a review @Gio2018, although there is one slight issue: an AutoLayout constraint that's broken when I hide the view on the Jetpack app.  I'll give this another pass to try to fix that but I wanted to submit this for review, to get your thoughts.

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    func setup() {
+
+        translatesAutoresizingMaskIntoConstraints = false
+        backgroundColor = Appearance.jetpackBackgroundColor
+
+        let jetpackButton = makeJetpackButton()
+        addSubview(jetpackButton)
+
+        pinSubviewToAllEdges(jetpackButton)
+        heightAnchor.constraint(equalToConstant: Appearance.jetpackBannerHeight).isActive = true
```
Comment (Gio2018) [inline WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift:29]: We could also remove this constraint here, and add `JEtpackBannerView` as a subview in the container view and apply the constraint externally, or we could expose an `height` property to be overridden: this to allow more flexibility on the view and not just assume that it has to have an height of 44. 
On the other hand, we don't need to do it on this PR, can be safely done in a future one.

**Review by Gio2018 — APPROVED** [review]
Thanks for this PR @Guarani !!

:shipit: !!

File: WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift (hunk 4 above)
Comment (guarani) [reply WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift:29]: > this to allow more flexibility on the view and not just assume that it has to have an height of 44.

Is this to allow the banner to expand in height if for example, the user increases text size?

Comment (guarani) [reply WordPress/Classes/ViewRelated/Jetpack Branding/JetpackBannerView.swift:29]: > On the other hand, we don't need to do it on this PR, can be safely done in a future one.

Ok, I won't consider this a blocker for this PR and will move ahead with getting it merged. We can continue this discussion here of course.

**Review by ovitrif — APPROVED** [review]
I tested this in the WP & JP apps and reviewed the code, it looks good to me 🎉 !

Great work both @Gio2018 & @guarani!

There's one small UI detail that I feel it's missing: the banner was designed with a small & slightly visible top border that appears to be missing in the implementation:

| Design | Implementation |
| --- | --- |
| stats_design | stats_impl |

I'll leave it up you to decide how to proceed as I don't have a strong opinion about whether it must be added in this PR or not.

**Comment by guarani** [comment]
Thank you for the review, @ovitrif!

I didn't notice the missing top border, thanks for flagging it! I'll merge this to lock it in, and I can add the top border (`border-top: 0.5px solid #DCDCDE;`) in a later PR, targeting 20.3.

Merging this now.

cc @mokagio

**Pull request merged**